### PR TITLE
Data augmentation

### PR DESCRIPTION
**Pull request by ChangLiu2022**
Seperated time history as two functions (2D and 3D)
Removed legacy functions.
Resolved scipy/sklearn confusion.
Added more detailed descriptions.
Added requirements for data augmentation
Add PyTorch implementation of StandardScaler (from @jtcostello 's code)

### CODE REVIEW

**Review by draelos — CHANGES_REQUESTED** [review]
Are those python packages in requirements.txt necessary versions? #32 

Function inputs in DataAugmentation are different for different functions -- inline or separate lines?

File: requirements.txt [hunk 1]
```diff
@@ -5,4 +5,6 @@ torch==2.4.0
 torchvision==0.19.0
 torchaudio==2.4.0
 pynwb==2.8.2
-neuroconv==0.6.5
\ No newline at end of file
+neuroconv==0.6.5
```
[the remark this answers is not in the document]
Comment (ChangLiu2022) [reply requirements.txt:8]: neuroconv is not used for data augmentation functions. It was already in requirements.txt before I added requirement for scipy and scikit-learn. It's probably marked as changed because of the new line.

Regarding scipy and scikit-learn version, it should work as long as the version used have function scipy.spatial.KDTree and sklearn.base.BaseEstimator.

Do you want the version for scipy and scikit-learn to be removed?

[the remark this answers is not in the document]
Comment (htemmar) [reply requirements.txt:8]: keeping requirement is fine - maybe chang to >= rather than ==. Also, if possible would add a test script for showing these changes working

**Review by htemmar — CHANGES_REQUESTED** [review]
style guide: classes should be CapWords, functions/methods should be lowercase with underscore between words. module names should be all lowercase and short. please update accordingly

noise tests seem good

need to add tests for scaler
- forward transform normalizes - maybe not needed

and add tests for other data aug methods

**Review by draelos — APPROVED** [review]
Looks good to merge in and keep iterating!
